src/java.base/share/classes/javax/crypto/JceSecurityManager.java

```diff
@@ -100,28 +105,19 @@ CryptoPermission getCryptoPermission(String alg) {
         // javax.crypto.* packages.
         // NOTE: javax.crypto.* package maybe subject to package
         // insertion, so need to check its classloader as well.
-        Class<?>[] context = getClassContext();
-        URL callerCodeBase = null;
-        int i;
-        for (i=0; i<context.length; i++) {
-            Class<?> cls = context[i];
-            callerCodeBase = JceSecurity.getCodeBase(cls);
-            if (callerCodeBase != null) {
-                break;
-            } else {
-                if (cls.getName().startsWith("javax.crypto.")) {
-                    // skip jce classes since they aren't the callers
-                    continue;
-                }
-                // use default permission when the caller is system classes
-                return defaultPerm;
-            }
-        }
-
-        if (i == context.length) {
-            return defaultPerm;
-        }
+        return WALKER.walk(s -> s.map(StackFrame::getDeclaringClass)
+                .filter(c -> !c.getPackageName().equals("javax.crypto"))
+                .map(cls -> {
+                    URL callerCodeBase = JceSecurity.getCodeBase(cls);
+                    return (callerCodeBase != null) ?
+                            getCryptoPermissionFromURL(callerCodeBase,
+                                    alg, defaultPerm) : defaultPerm;})
```

Comment:
nit: the line break at ":" would make it easier to read than breaking at `arg`.

Reply:
Adding all three params would make it > 80 char lines.